src/v/kafka/client/client.cc

```diff
@@ -350,7 +350,40 @@ ss::future<fetch_response> client::fetch_partition(
                            return _brokers.find(leader);
                        })
                        .then([&tp, &build_request](shared_broker_t&& b) {
-                           return b->dispatch(build_request(tp));
+                           return b->dispatch(build_request(tp))
+                             .then([b, &tp](fetch_response res) {
+                                 if (res.data.error_code != error_code{}) {
+                                     return ss::make_exception_future<
+                                       fetch_response>(broker_error(
+                                       b->id(), res.data.error_code));
+                                 }
+
+                                 if (res.data.topics.size() < 1) {
```

Comment:
nit: `.size() < 1` can be spelled `!.empty()`

The request only has a single topic-partition, so the loops can probably be removed in favour of a check of `.size() == 1`.

Reply:
Done.